Add com.github.matfantinel.reminduck

matfantinel opened the pull request.

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,72 @@
+       "--talk-name=org.gtk.vfs.*",
+       "--talk-name=ca.desrt.dconf",
+       "--filesystem=home",
+       "--filesystem=xdg-run/dconf",
```
You should remove dconf access. It's not needed anymore starting from FDO Sdk 19.08/GNOME Sdk 3.34

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,72 @@
+          "sources": [
+              {
+                  "type": "git",
+                  "url": "https://github.com/yaml/libyaml.git",
```
Use a tarball or add a commit along with the tag. The same for the other modules.

bilelmoussaoui replied:
Please don't mark conversations as resolved unless they are.

bilelmoussaoui commented:
bot, build com.github.matfantinel.reminduck

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,72 @@
+          "sources": [
+              {
+                  "type": "git",
+                  "url": "https://github.com/matfantinel/reminduck.git",
```
You probably want to patch the application to use the elementary GTK & icons themes. They are already in the BaseApp.

matfantinel replied:
I've tested it with many different themes in a Gnome environment and am happy with the result. I prefer it to respect GTK theming!

bilelmoussaoui commented:
The build fails due to missing OARS tag in the appdata file https://hughsie.github.io/oars/generate.html

matfantinel commented:
Sorry @bilelmoussaoui , I've updated the tags and the manifest now.

LeandroStanger commented:
bot, build com.github.matfantinel.reminduck

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,70 @@
+    "finish-args": [
+       "--share=ipc",
+       "--share=network",
+       "--socket=x11",
```
```suggestion
       "--socket=fallback-x11",
```

bilelmoussaoui commented:
bot, build com.github.matfantinel.reminduck

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,70 @@
+                  "tag": "v0.12.9",
+                  "commit": "d77cc0f331ff5d3bd43be104106a94ed201c62ec"
+              }
+          ]
```
Include lmdb as a submodule of appstream https://github.com/flathub/io.github.NhekoReborn.Nheko/blob/master/io.github.NhekoReborn.Nheko.json#L28

bilelmoussaoui commented:
bot, build com.github.matfantinel.reminduck

bilelmoussaoui commented:
I have tried the app and noticed that two icons are missing from the default icon theme
![Screenshot from 2020-01-10 23-45-12](https://user-images.githubusercontent.com/7660997/72191694-5e4ef300-3403-11ea-9c0a-22d6460bba6f.png)

![Screenshot from 2020-01-10 23-44-55](https://user-images.githubusercontent.com/7660997/72191700-61e27a00-3403-11ea-8ea9-7ad3997c9320.png)

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,87 @@
+       "--socket=fallback-x11",
+       "--socket=wayland",
+       "--talk-name=org.gtk.vfs.*",
+       "--filesystem=home"
```
Couldn't find why you would need home access, network or vfs?

bilelmoussaoui commented on com.github.matfantinel.reminduck.json:
```diff
@@ -0,0 +1,87 @@
+{
+    "app-id": "com.github.matfantinel.reminduck",
+    "runtime": "org.gnome.Platform",
```
Any reason for not using the freedesktop runtime here?

matfantinel replied:
I have based my manifest on another app, and don't really know the difference between the two

bilelmoussaoui replied:
All the runtimes/sdk's are based on freedesktop runtime/sdk. The GNOME runtime/sdk contains libraries and tools specific to GNOME like Vala. As you're already using the elementary baseApp which contains vala + granite & the elementary stylesheet & icons, you can just switch to the freedesktop runtime.

matfantinel commented:
About the missing icons... I suppose the best way to deal with that is by forcing the elementary icons within the app like you mentioned. I'll have to research on how to do that and will update this PR when possible.

Thanks a lot for the feedback!

Exalm commented:
Not trying to persuade, but there are also some issues with the style. Like look at header labels, which use the default style because Adwaita doesn't have granite style classes. Or the headerbar height, and you can see the back button is awkwardly aligned. (that's caused by `default-decoration` style class FTR). Basically things like this is why elementary apps are often suggested to hardcode stylesheet, as trying to support both is often a huge amount of work.

matfantinel commented:
Thanks everyone for your input. I have updated the application and manifest to use the freeDesktop SDK and runtime, removed unnecessary permissions, and patched in the elementary theme. I tested it locally and all boxes seemed to check.

LeandroStanger commented:
bot, build com.github.matfantinel.reminduck

bilelmoussaoui commented:
/merge
